apps/antalmanac/src/components/RightPane/Map/UCIMap.tsx

```diff
@@ -107,27 +114,62 @@ export default class UCIMap extends PureComponent {
                                 courses.has(event.sectionCode) || // Remove duplicate courses that appear in the calendar
                                 !courses.add(event.sectionCode)
                             ) // Adds to the set and return false
+                            &&
+                            (
+                                !event.isCustomEvent ||
+                                !event.start.toString().includes(DAYS[day])
+                            )
                         )
                 )
                 .sort((event, event2) => event.start.getTime() - event2.start.getTime())
                 .forEach((event) => {
-                    if (event.isCustomEvent) return;
-                    // Get building code, get id of building code, which will get us the building data from buildingCatalogue
-                    const buildingCode = event.bldg.split(' ').slice(0, -1).join(' ') as keyof typeof locations;
-                    const id = locations[buildingCode] as keyof typeof buildingCatalogue;
-                    const locationData = buildingCatalogue[id];
+                    if (event.isCustomEvent) {
+                        // Get building code, get id of building code, which will get us the building data from buildingCatalogue
+                        var buildingCode;
+                        var id;
+                        var locationData;
+                        if (event.bldg.includes("(")){
```

Comment:
I don't see why a lot of this stuff is specific to custom events. Is there any way we can reuse this stuff between both cases (custom and non-custom events) so that the stuff inside `if(event.isCustomEvent)` is very clearly only stuff that custom events need? And on that note, how are the custom event markers even different than the normal markers? Can we just use the same logic for both and not even have that conditional?

Reply:
Yes, for this file, I can put most of the same code together.